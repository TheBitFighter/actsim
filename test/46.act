defchan e1of1 <: chan(enum<1>) (bool r; bool a)
{
  methods {
    set {
	r+
    }
    send_rest {
    	[a];r-;[~a]
    }
    get {
	[r]
    }
    send_init {
        r-
    }
    recv_rest {
        a+;[~r];a-
    }
    recv_probe = r;
  }
}


defproc src (e1of1 x)
{
  int i;
  chp {
     i := 0;
    *[ i < 100 -> x!; i := i + 1; log("sent!") ]
  }
}

defproc sink (e1of1 x)
{
  prs {
    x.r -> x.a+
   ~x.r -> x.a-
  }
}

defproc test ()
{
   src s;
   sink t(s.x);
}

