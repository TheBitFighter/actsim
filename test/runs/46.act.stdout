[                  60] <s>  sent!
[                 100] <s>  sent!
[                 140] <s>  sent!
[                 180] <s>  sent!
[                 220] <s>  sent!
[                 260] <s>  sent!
[                 300] <s>  sent!
[                 340] <s>  sent!
[                 380] <s>  sent!
[                 420] <s>  sent!
[                 460] <s>  sent!
[                 500] <s>  sent!
[                 540] <s>  sent!
[                 580] <s>  sent!
[                 620] <s>  sent!
[                 660] <s>  sent!
[                 700] <s>  sent!
[                 740] <s>  sent!
[                 780] <s>  sent!
[                 820] <s>  sent!
[                 860] <s>  sent!
[                 900] <s>  sent!
[                 940] <s>  sent!
[                 980] <s>  sent!
[                1020] <s>  sent!
[                1060] <s>  sent!
[                1100] <s>  sent!
[                1140] <s>  sent!
[                1180] <s>  sent!
[                1220] <s>  sent!
[                1260] <s>  sent!
[                1300] <s>  sent!
[                1340] <s>  sent!
[                1380] <s>  sent!
[                1420] <s>  sent!
[                1460] <s>  sent!
[                1500] <s>  sent!
[                1540] <s>  sent!
[                1580] <s>  sent!
[                1620] <s>  sent!
[                1660] <s>  sent!
[                1700] <s>  sent!
[                1740] <s>  sent!
[                1780] <s>  sent!
[                1820] <s>  sent!
[                1860] <s>  sent!
[                1900] <s>  sent!
[                1940] <s>  sent!
[                1980] <s>  sent!
[                2020] <s>  sent!
[                2060] <s>  sent!
[                2100] <s>  sent!
[                2140] <s>  sent!
[                2180] <s>  sent!
[                2220] <s>  sent!
[                2260] <s>  sent!
[                2300] <s>  sent!
[                2340] <s>  sent!
[                2380] <s>  sent!
[                2420] <s>  sent!
[                2460] <s>  sent!
[                2500] <s>  sent!
[                2540] <s>  sent!
[                2580] <s>  sent!
[                2620] <s>  sent!
[                2660] <s>  sent!
[                2700] <s>  sent!
[                2740] <s>  sent!
[                2780] <s>  sent!
[                2820] <s>  sent!
[                2860] <s>  sent!
[                2900] <s>  sent!
[                2940] <s>  sent!
[                2980] <s>  sent!
[                3020] <s>  sent!
[                3060] <s>  sent!
[                3100] <s>  sent!
[                3140] <s>  sent!
[                3180] <s>  sent!
[                3220] <s>  sent!
[                3260] <s>  sent!
[                3300] <s>  sent!
[                3340] <s>  sent!
[                3380] <s>  sent!
[                3420] <s>  sent!
[                3460] <s>  sent!
[                3500] <s>  sent!
[                3540] <s>  sent!
[                3580] <s>  sent!
[                3620] <s>  sent!
[                3660] <s>  sent!
[                3700] <s>  sent!
[                3740] <s>  sent!
[                3780] <s>  sent!
[                3820] <s>  sent!
[                3860] <s>  sent!
[                3900] <s>  sent!
[                3940] <s>  sent!
[                3980] <s>  sent!
[                4020] <s>  sent!
